### PR TITLE
Migrate user permissions into grants

### LINKED ISSUE #8919: Migrate user permissions into grants
The currently existing permissions that are directly stored within the user collection
should also be migrated to grants (where applicable)

The conversion should be done for Dashboards and Streams:
View + Edit  -> Manage 
View             -> Viewer

### PR DESCRIPTION
**Pull request by mpfz0r**
Fixes #8919

/jenkins-pr-deps Graylog2/graylog-plugin-enterprise#1569

### CODE REVIEW

**Review by kmerz — APPROVED** [review]
Code LGTM :+1: 
Worked on my machine :heavy_check_mark: 

wrong browser tab

**Review by kmerz — CHANGES_REQUESTED** [review]
Did not work on my machine!

**Review by kmerz — APPROVED** [review]
Code LGTM :+1: 
Tested with different configurations and it worked :heavy_check_mark:

File: graylog2-server/src/main/java/org/graylog/security/DBGrantService.java [hunk 1]
```diff
@@ -136,6 +141,27 @@ public GrantDTO create(GRN grantee, Capability capability, GRN target, String cr
...
+        final List<GrantDTO> existingGrants = getForTargetAndGrantee(target, grantee);
+        if (existingGrants.isEmpty()) {
+            return create(grantee, capability, target, creatorUsername);
+        }
+        // This should never happen
+        Preconditions.checkState(existingGrants.size() == 1);
+
+        final GrantDTO grantDTO = existingGrants.get(0);
+        // Only upgrade capabilities: VIEW < MANAGE < OWNER
+        if (capability.ordinal() > grantDTO.capability().ordinal()) {
```
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog/security/DBGrantService.java:158]: I am not sure if using the #ordinal method of the enum will bite us at one point in the future. The value will change if the enums are reordered at some point. (e.g. someone wants them to be sorted alphabetically and refactors the order) How about adding an explicit priority value to each enum instead?

**Review by bernd — CHANGES_REQUESTED** [review]
Only one comment:

File: graylog2-server/src/main/java/org/graylog/security/DBGrantService.java (hunk 1 above)
Comment (mpfz0r) [reply graylog2-server/src/main/java/org/graylog/security/DBGrantService.java]: fair enough. :+1:

**Review by bernd — APPROVED** [review]

**Pull request merged**